lib/services/security/service.js

```diff
@@ -843,7 +848,7 @@ function onDataChanged(hook) {
 function getEffectedSession(sessionData, causeSubscriptionsRefresh) {
   return {
     id: sessionData.id,
-    username: sessionData.user.username,
+    username: sessionData.user?sessionData.user.username:'unknown',

```

Comment:
i know it is unlikely but what effect will it have if a user makes his username 'unknown'?

Reply:
should I make it 'undefined'? not much will happen, except that the 'unknown' username will be logged downstream in exactly the same way

Reply:
If it won't have any unintended consequences it is fine to leave as is.